test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+    (
+      SessionData.empty.copy(
+        journeyStatus = Some(journey)
+      ),
+      journey,
+      draftReturn
+    )
+  }
+
+  def testFormError(
+    data: (String, String)*
+  )(expectedErrorMessageKey: String, errorArgs: Seq[String] = Nil)(pageTitleKey: String, titleArgs: String*)(
+    performAction: Seq[(String, String)] => Future[Result],
+    currentSession: SessionData = sessionWithSingleDisposalState(
+      Some(sample[CompleteSupportingEvidenceAnswers])
+    )._1
+  ): Unit = {
+    inSequence {
+      mockAuthWithNoRetrievals()
+      mockGetSession(currentSession)
+    }
+    checkPageIsDisplayed(
+      performAction(data),
+      messageFromMessageKey(pageTitleKey, titleArgs: _*), { doc =>
+        doc.select("#error-summary-display > ul > li > a").text() shouldBe messageFromMessageKey(
+          expectedErrorMessageKey,
+          errorArgs: _*
+        )
+        doc.title() should startWith("Error:")
+      },
+      BAD_REQUEST
+    )
+  }
+
+  "Supporting Evidence Controller" when {
+
+    "handling requests to display the do you want to upload supporting evidence page" must {
+
+      def performAction(): Future[Result] = controller.doYouWantToUploadSupportingDocuments()(FakeRequest())
+
+      "display the page" when {
+
+        "a user has not answered the question before" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithSingleDisposalState(
+                None
+              )._1
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey("supporting-evidence.do-you-want-to-upload.title"), { doc =>
+              doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.SupportingEvidenceController
+                .doYouWantToUploadSupportingDocuments()
+                .url
+            }
+          )
+        }
+
+        "a user has answered the question before" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithSingleDisposalState(
+                Some(IncompleteSupportingEvidenceAnswers(Some(false), List.empty, List.empty))
+              )._1
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey("supporting-evidence.do-you-want-to-upload.title"), { doc =>
+              doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.SupportingEvidenceController
+                .doYouWantToUploadSupportingDocuments()
+                .url
+            }
```

Comment:
if you wanted to check that the option pre-selected here when the user has already answered the question before you could do something like:
```scala
doc.select("#{no-id)").attr("checked") shouldBe "checked"
```